src/lib/marshal/translation/processor.isomorphic.ts

```diff
@@ -37,20 +44,26 @@ type OneOrMoreTranslationData = TranslationData | TranslationData[];
 
 export const buildTranslationDirBundle = (
   input: OneOrMoreTranslationData,
+  options?: {
+    format?: TranslationFormat;
+  },
 ): TranslationDirBundle => {
+  const format = options?.format ?? DEFAULT_TRANSLATION_FORMAT;
   if (Array.isArray(input)) {
     const translations = input;
 
     return Object.fromEntries(
       translations.map((translation) => [
-        formatFileName(translation),
+        formatFileName(translation, { format }),
         JSON.parse(translation.content),
       ]),
     );
   }
 
   const translation = input;
+  const content =
+    format === "po" ? translation.content : JSON.parse(translation.content);
```

Comment:
Can we pls flip this conditional so that it does "if json, then do the json specific thing.."?